light/txpool.go

```diff
@@ -278,7 +278,7 @@ func (pool *TxPool) setNewHead(ctx context.Context, newHeader *types.Header) (tx
 	if idx := newHeader.Number.Uint64(); idx > pool.clearIdx+txPermanent {
 		idx2 := idx - txPermanent
 		for i := pool.clearIdx; i < idx2; i++ {
-			hash := core.GetCanonicalHash(pool.chainDb, i)
+			hash := pool.chain.GetStoredCanonicalHash(i)
```

Comment:
I removed the whole caching thing because I realized that it would add too much complexity and probably solve a non-existent problem.